lib/ibandit/iban.rb

```diff
@@ -186,13 +186,14 @@ def decomposable?
     end
 
     def build_iban_from_local_details(details_hash)
-      local_details = details_hash.dup
+      local_details = LocalDetailsCleaner.clean(details_hash)
```

Comment:
The cleaner always does a `details_hash.merge(cleaned_stuff)`, so will return a different hash. The individual strings within in could still get mutated though, which would be annoying. I'll `.dup` them all.